src/components/table/table.types.ts

```diff
@@ -37,6 +37,8 @@ export interface TableComponentDefinition {
      * Properties to send to the component
      */
     props?: Record<string, any>;
+
+    propFactory?: (data: object) => Record<string, any>;
```

Comment:
We need to add documentation here so people know how to use it :slightly_smiling_face:

Reply:
We might also want to rename this to `propsFactory` to match the `props` property